modules/openthread/platform/spinel_hdlc.hpp

```diff
@@ -0,0 +1,207 @@
+
+    /**
+     * This method deinitializes the HDLC interface.
+     *
+     */
+    void Deinit(void);
+
+    /**
+     * This method performs radio driver processing.
+     *
+     * @param[in]  aInstance  The ot instance
+     *
+     */
+    void Process(const void *aInstance);
+
+    /**
+     * This method encodes and sends a spinel frame to Radio Co-processor (RCP) over the socket.
+     *
+     * This is blocking call, i.e., if the socket is not writable, this method waits for it to become writable for
+     * up to `kMaxWaitTime` interval.
+     *
+     * @param[in] aFrame     A pointer to buffer containing the spinel frame to send.
+     * @param[in] aLength    The length (number of bytes) in the frame.
+     *
+     * @retval OT_ERROR_NONE     Successfully encoded and sent the spinel frame.
+     * @retval OT_ERROR_NO_BUFS  Insufficient buffer space available to encode the frame.
+     * @retval OT_ERROR_FAILED   Failed to send due to socket not becoming writable within `kMaxWaitTime`.
+     *
+     */
+    otError SendFrame(const uint8_t *aFrame, uint16_t aLength);
+
+    /**
+     * This method waits for receiving part or all of spinel frame within specified timeout.
+     *
+     * @param[in]  aTimeoutUs  The timeout value in microseconds.
+     *
+     * @retval OT_ERROR_NONE             Part or all of spinel frame is received.
+     * @retval OT_ERROR_RESPONSE_TIMEOUT No spinel frame is received within @p aTimeoutUs.
+     *
+     */
+    otError WaitForFrame(uint64_t aTimeoutUs);
+
+    /**
+     * This method is called by the HDLC RX Callback when a HDLC message has been received
+     *
+     * It will decode and store the Spinel frames in a temporary Spinel frame buffer (mRxSpinelFrameBuffer)
+     * This buffer will be then copied to the OpenThread Spinel frame buffer, from the OpenThread task context
+     *
+     * @param[in] data A pointer to buffer containing the HDLC message to decode.
+     * @param[in] len  The length (number of bytes) in the message.
+     */
+    void ProcessRxData(uint8_t *data, uint16_t len);
+
+    /**
+     * This method is called when RCP failure detected and resets internal states of the interface.
+     *
+     */
+    void OnRcpReset(void);
+
+    /**
+     * This method is called when RCP is reset to recreate the connection with it.
+     * Intentionally empty.
+     *
+     */
+    otError ResetConnection(void) { return OT_ERROR_NONE; }
+
+    /**
+     * This method hardware resets the RCP.
+     *
+     * @retval OT_ERROR_NONE             Successfully reset the RCP.
+     * @retval OT_ERROR_NOT_IMPLEMENTED  The hardware reset is not implemented.
+     *
+     */
+    otError HardwareReset(void) { return OT_ERROR_NOT_IMPLEMENTED; }
+
+private:
+
+	/* Spinel HDLC interface */
+    const struct device *radio_dev;
+    struct hdlc_api     *hdlc_api;
+
+    struct k_event  s_spinel_hdlc_event;
```

Comment:
A general comment about coding style - there seems to be a mixup of two coding styles, OpenThread's `camelCase`, and Zephyrs `snake_case`. As a general rule, for code implemented on the Zephyr side, we should stick to Zephyr's CS, unless unavoidable (I. e. function prototypes defined in OT etc.). I am aware there are some deviations from this rule, due to past oversights, but let's not make them an argument for not following Zephyr's CS.

Reply:
I will have a look on the coding style.